pyomo/contrib/fbbt/interval.py

```diff
@@ -34,7 +38,11 @@ def power(xl, xu, yl, yu):
         elif yu <= 0:
             lb = xu ** yl
             ub = xl ** yu
+        else:
+            raise RuntimeError('This was unexpected.')
```

Comment:
I would replace this with a DeveloperError -- that is, an exception that you would never expect to be able to be hit, and if someone does, they should report it to the developers.

Reply:
Good suggestion. I will fix this.